dodiscover/constraint/intervention.py

```diff
@@ -0,0 +1,362 @@
+        The method to use for selecting the conditioning sets using PDS. Must be one of
+        ('pds', 'pds_path'). See Notes for more details.
+    known_intervention_targets : bool, optional
+        If `True`, then will run the I-FCI algorithm. If `False`, will run the
+        Psi-FCI algorithm. By default False.
+
+    Notes
+    -----
+    Selection bias is unsupported because it is still an active research area.
+    """
+
+    def __init__(
+        self,
+        ci_estimator: BaseConditionalIndependenceTest,
+        cd_estimator: BaseConditionalDiscrepancyTest,
+        alpha: float = 0.05,
+        min_cond_set_size: Optional[int] = None,
+        max_cond_set_size: Optional[int] = None,
+        max_combinations: Optional[int] = None,
+        condsel_method: ConditioningSetSelection = ConditioningSetSelection.NBRS,
+        apply_orientations: bool = True,
+        keep_sorted: bool = False,
+        max_iter: int = 1000,
+        max_path_length: Optional[int] = None,
+        pds_condsel_method: ConditioningSetSelection = ConditioningSetSelection.PDS,
+        known_intervention_targets: bool = False,
+        n_jobs: Optional[int] = None,
+    ):
+        super().__init__(
+            ci_estimator,
+            alpha,
+            min_cond_set_size,
+            max_cond_set_size,
+            max_combinations,
+            condsel_method,
+            apply_orientations,
+            keep_sorted=keep_sorted,
+            max_iter=max_iter,
+            max_path_length=max_path_length,
+            selection_bias=False,
+            pds_condsel_method=pds_condsel_method,
+            n_jobs=n_jobs,
+        )
+        self.cd_estimator = cd_estimator
+        self.known_intervention_targets = known_intervention_targets
+
+    def learn_skeleton(
+        self, data: pd.DataFrame, context: Context, sep_set: Optional[SeparatingSet] = None
+    ) -> Tuple[nx.Graph, SeparatingSet]:
+        # now compute all possibly d-separating sets and learn a better skeleton
+        self.skeleton_learner_ = LearnInterventionSkeleton(
+            self.ci_estimator,
+            self.cd_estimator,
+            sep_set=sep_set,
+            alpha=self.alpha,
+            min_cond_set_size=self.min_cond_set_size,
+            max_cond_set_size=self.max_cond_set_size,
+            max_combinations=self.max_combinations,
+            condsel_method=self.condsel_method,
+            second_stage_condsel_method=self.pds_condsel_method,
+            keep_sorted=False,
+            max_path_length=self.max_path_length,
+            n_jobs=self.n_jobs,
+        )
+        self.skeleton_learner_.fit(data, context)
+
+        self.context_ = self.skeleton_learner_.context_.copy()
+        skel_graph = self.skeleton_learner_.adj_graph_
+        sep_set = self.skeleton_learner_.sep_set_
+        self.n_ci_tests += self.skeleton_learner_.n_ci_tests
+        return skel_graph, sep_set
+
+    def fit(self, data: List[pd.DataFrame], context: Context):
```

Comment:
Currently, it leverages fit because presumably someone could restart the process (e.g. AnytimeFCI) at any point. However, to enable this feature robustly seems like quite a bit of work. 

I think Robert suggested we use the word `learn_graph`, or something along those lines. Regardless, I think the keyword to use could use some discussion and I'm happy to change them all simultaneously across the board once we reach a consensus.

Reply:
I like `learn_graph` but am open to other ideas.

Reply:
Yeah that word def sounds more in line. I can then change it to learn_graph in a follow on PR?